### PR TITLE
Fixing hanging Ice callback

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
1. hanging web

 > For each vhost Django site should use a separate daemon process group. WSGIApplicationGroup %{GLOBAL} https://code.google.com/p/modwsgi/wiki/ConfigurationDirectives#WSGIApplicationGroup is forcing the use of the main interpreter context within the respective process groups. It is not shared across process groups.

 Looks like we are using Python modules (IcePy?) that are not loaded correctly in Python sub interpreters, see https://code.google.com/p/modwsgi/wiki/ApplicationIssues#Python_Simplified_GIL_State_API

 To test:
 - delete image and check activities, it shouldn't hang anymore.

1. I also updated python-path as it was missing env, cc @kennethgillen 
 To test:
 -  default ``PYTHONPATH``
    - set ``bin/omero config set omero.web.application_server 'wsgi'``
    - make sure there is nothing in your pythonpath ``$echo $PYTHONAPTH``
    - generate config ``bin/omero web config {apache22|apache24}``
    - ``WSGIDaemonProcess`` should contain only  defaults in ``python-path=..python2.7/site-packages:..dist/lib/python:...``
  - custom PYTHONPATH:
    - set ``bin/omero config set omero.web.application_server 'wsgi'``
    - export ``$PYTHONPATH=/some/location:$PYTHONPATH``
    - generate config ``bin/omero web config {apache22|apache24}``
    - check if ``WSGIDaemonProcess`` contains ``/some/location`` in ``python-path=..python2.7/site-packages:..dist/lib/python:/some/location``

**Note: for multiple WSGI applications, config delegate each WSGI application to run in its own daemon process group.** See https://groups.google.com/forum/m/#!topic/modwsgi/7NRGTm4VUp4

### CODE REVIEW

**Comment by joshmoore** [comment]
Nice find, @aleksandra-tarkowska ! Perhaps now the question is there will be any performance impact...

**Comment by aleksandra-tarkowska** [comment]
@joshmoore performance can be altered by number of processes and threads https://code.google.com/p/modwsgi/wiki/ProcessesAndThreading

**Comment by aleksandra-tarkowska** [comment]
Tested externally by https://www.openmicroscopy.org/community/viewtopic.php?f=5&t=7954&p=16522#p16522

**Comment by will-moore** [comment]
Got quite a long way with Apache setup etc, but am still getting this in the error log, and haven't managed to find a fix yet.
```
[Wed Dec 02 12:00:16.137350 2015] [wsgi:error] [pid 88757] (2)No such file or directory: [client ::1:55941] mod_wsgi (pid=88757): Unable to connect to WSGI daemon process 'omeroweb' on '/usr/run/wsgi.88109.0.1.sock' as user with uid=70., referer: http://localhost/
```

**Comment by aleksandra-tarkowska** [comment]
did you ``chmod a+X``

**Comment by aleksandra-tarkowska** [comment]
@will-moore you need to change to WSGISocketPrefix /var/run/wsgi, see last line of the config

**Comment by will-moore** [comment]
I fixed a previous permissions issue with OMERO dir by doing:
```
chmod 755 /Users/wmoore/Desktop
```
Is there another directory path that I need to check permissions on?

**Comment by will-moore** [comment]
OK, that fixed it, thanks.
Delete, callbacks etc is working.
$PYTHONPATH is included in WSGIDaemonProcess python-path.

Good to merge.

File: components/tools/OmeroPy/src/omero/plugins/web.py [hunk 1]
```diff
@@ -254,7 +254,13 @@ def _set_apache_wsgi(self, d, settings):
             print traceback.print_exc()
             self.ctx.err(
                 "Cannot import Ice.")
-        d["OMEROPYTHONROOT"] = self._get_python_dir()
+        try:
+            pythopath = ":".join([
```
Comment (joshmoore) [inline components/tools/OmeroPy/src/omero/plugins/web.py]: Will this ever get called on Windows? If so, I'd suggest:
```
>>> os.pathsep
':'
```

Comment (mtbc) [inline components/tools/OmeroPy/src/omero/plugins/web.py]: (It's not called `pythonpath` for a good reason I assume.)

Comment (aleksandra-tarkowska) [inline components/tools/OmeroPy/src/omero/plugins/web.py]: I haven;t seen anyone trying to use Apache on Windows, sure will fix it, thx

Comment (aleksandra-tarkowska) [inline components/tools/OmeroPy/src/omero/plugins/web.py]: @mtbc ;-)

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-DEV-merge-push#175](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-push/175/). See the [console output](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-push/175/consoleText) for more details.
Possible conflicts:
  - PR #4352 aleksandra-tarkowska 'wsgi-args in config'
    - components/tools/OmeroPy/test/unit/clitest/test_web.py

**Comment by aleksandra-tarkowska** [comment]
@joshmoore Could this PR be merged please? I will rebase #4352 then

**Comment by jburel** [comment]
@aleksandra-tarkowska: did you relist this PR to test your last commit? before we can merge it
I cannot see it standup 
Thanks

**Comment by aleksandra-tarkowska** [comment]
I thought passing unit test was enough. This is only about separator right?

**Comment by will-moore** [comment]
Testing last commit, concatenation of PYTHONPATHs:
```
$ echo $PYTHONPATH 
/Users/wmoore/Desktop/FIGURE:/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/python:

$ bin/omero web config apache24
...
 WSGIDaemonProcess omeroweb processes=5 threads=1 display-name=%{GROUP} user=wmoore python-path=/usr/local/lib/python2.7/site-packages:/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/python:/Users/wmoore/Desktop/FIGURE:/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/python::/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/fallback:/Users/wmoore/Desktop/OMERO/openmicroscopy/dist/lib/python/omeroweb
...
```
Working as expected. Good to merge.

**Comment by jburel** [comment]
Thanks for testing again @will-moore

**Pull request merged**